src/ParamConverter/SparqlEntityStorageConverter.php

```diff
@@ -84,6 +87,10 @@ public function convert($value, $definition, $name, array $defaults) {
     if ($entity instanceof EntityInterface && $entity instanceof TranslatableInterface) {
       $entity = $this->entityRepository->getTranslationFromContext($entity, NULL, ['operation' => 'entity_upcast']);
     }
+
+    if (!empty($definition['bundle']) && !in_array($entity->bundle(), $definition['bundle'], TRUE)) {
+      return NULL;
+    }
```

Comment:
This does not check if `$entity` is an object of type `EntityInterface` and throws a fatal error when `$entity` is `NULL`.